### PR TITLE
change veracode_application to veracode_assessment

### PR DESCRIPTION
**Pull request by mdaum**
# Description
change veracode_application to veracode_assessment

## Summary
- change veracode_application to veracode_assessment
- change class of the aforementioned entity from Application to Assessment
- make necessary edits throughout project to jive with this change

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [X] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [X] My changes properly paginate the target service provider's API
- [X] My changes properly handle rate limiting of the target service provider's
      API
- [X] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [X] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [X] I have updated the `CHANGELOG.md` file to describe my changes
- [X] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export interface IntegrationConfig extends IntegrationInstanceConfig {
   /**
    * For use in tests only. Avoid ingesting entire account's worth of applications to reduce PollyJS Recording size
    */
-  targetApplication?: string;
+  targetVeracodeApplication?: string;
```
Comment (adam-in-ict) [inline src/config.ts:52]: nit:  Did you mean to change this and the above comment to read 'assessment' instead?

File: src/steps/assessment/converter.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  createIntegrationEntity,
+  Entity,
+  parseTimePropertyValue,
+} from '@jupiterone/integration-sdk-core';
+import { Application } from '../../types';
+
+import { Entities } from '../constants';
+
+export function createAssessmentEntity(
+  veracodeApplication: Application,
```
Comment (adam-in-ict) [inline src/steps/assessment/converter.ts:11]: Did you mean for this to still be called an Application throughout this file?

File: test/config.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ const DEFAULT_API_SECRET = 'dummy-veracode-secret';
 export const integrationConfig: IntegrationConfig = {
   apiId: process.env.API_ID || DEFAULT_API_ID,
   apiSecret: process.env.API_SECRET || DEFAULT_API_SECRET,
-  targetApplication: 'US1492-1',
+  targetVeracodeApplication: 'US1492-1',
```
Comment (adam-in-ict) [inline test/config.ts:19]: nit:  Did you mean to change this to be an Assessment?

File: src/config.ts (hunk 1 above)
Comment (mdaum) [reply src/config.ts:52]: In the [veracode api](https://app.swaggerhub.com/apis/Veracode/veracode-applications_api_specification/1.0#/Application%20information%20API/getApplicationsUsingGET) we are pulling in what they denote as an `Application`. However, that entity class is primarily used in J1 as an anchor point in the graph connecting various subgraphs sourced from different integration definitions.

So throughout the project when I refer to Application, I am specifically mentioning the VeracodeApplication as represented in their API. When we eventually represent this data from Veracode as an Entity with `_class=Assessment`, we switch to the `Assessment` naming convention.

Comment (mdaum) [reply src/config.ts:52]: I allude to this [here](https://github.com/JupiterOne/graph-veracode/pull/20/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR1), but I am happy to leave a more descriptive comment in that file over the `Application` type. Which should perhaps be renamed to `VeracodeApplication` now that I think more about it.

**Review by adam-in-ict — APPROVED** [review]

**Pull request merged**